src/main/haskell/kore/src/Kore/Step/Step.hs

```diff
@@ -118,7 +120,12 @@ transitionRule tools substitutionSimplifier simplifier =
             $ stepWithAxiom tools substitutionSimplifier config a
         case result of
             Left _ -> pure []
-            Right (config', proof') ->
+            Right
+                ( StepResult
+                    { rewrittenPattern = config'
+                    -- TODO(virgil): Also use the remainder
```

Comment:
If `StepResult` is equivalent to `OrOfExpandedPattern`, we would only need to extract and return the list below.

Reply:
As mentioned above, even if it is so, we might want to do this only when doing proofs.